Libplanet.Action/ActionEvaluator.cs

```diff
@@ -324,6 +313,18 @@ IActionContext CreateActionContext(IAccount newPrevState)
 
             state = feeCollector.Refund(state);
             state = feeCollector.Reward(state);
+            state = state is Account a
+                ? new Account(
+                    new AccountState(stateStore.Commit(a.Trie)),
+                    a.TotalUpdatedFungibles)
+                : throw new InvalidOperationException(
+                    $"Internal {nameof(IAccount)} is not of valid type: {state.GetType()}");
+
+            if (!state.Trie.Recorded)
+            {
+                throw new InvalidOperationException(
+                    $"Failed to record {nameof(IAccount)}'s {nameof(ITrie)}.");
+            }
```

Comment:
Does `stateStore.Commit(a.Trie)` not throw anything if the operation fails?

Reply:
Probably not, but I'm not so sure about all possible edge cases. 😥
Although, there is at least one I can come up with.
If `trie.Recorded` was `false` in the first place and it hasn't changed, `stateStore.Commit(trie)` will do nothing
and return an unrecorded `ITrie` instance. 😶

Reply:
Then again, the more I think about it, as `ActionEvaluator` has no control over what happens inside `IAction.Execute()`, I think this might be necessary. 😑

Reply:
There are also other issues, such as `ITrie` having to be internally coupled with `IStateStore` instance (more specifically, `IKeyValueStore` instance held internally by `IStateStore`) and the method being `static`, etc. etc. I'd say we leave it as it is for now...?

Reply:
Ok, I understood.